Microsoft.Toolkit/Diagnostics/Guard.String.cs

```diff
@@ -0,0 +1,233 @@
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must not be <see langword="null"/> or empty.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is <see langword="null"/> or empty.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNotNullOrEmpty(string? text, string name)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsNotNullOrEmpty(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must be <see langword="null"/> or whitespace.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is neither <see langword="null"/> nor whitespace.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNullOrWhitespace(string? text, string name)
+        {
+            if (!string.IsNullOrWhiteSpace(text))
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsNullOrWhitespace(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must not be <see langword="null"/> or whitespace.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is <see langword="null"/> or whitespace.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNotNullOrWhitespace(string? text, string name)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsNotNullOrWhitespace(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must be empty.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is empty.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsEmpty(string text, string name)
+        {
+            if (text.Length != 0)
```

Comment:
Shouldn't we also be worried about null here?

Reply:
That's not needed, as the entire class is using nullable reference types 😄

All the methods that accept `null` parameters are marked with the nullable attribute, and the others leave that check to the user. This is mostly done for clarity and for performance reasons, as otherwise we'd have to add redundant null checks basically everywhere.

Eg. if you're working on a `string` that could be `null`, you'd just need to call `Guard.IsNullOrEmpty(string, string)` instead of just `Guard.IsEmpty(string, string).`

Besides, if you passed a `null` value there you'd still end up with the same `NullReferenceException` anyway, so the actual code execution would stay the same.